src/rgw/rgw_sal.h

```diff
@@ -992,7 +992,7 @@ class Object {
     virtual void set_obj_state(RGWObjState& _state) = 0;
     /** Set attributes for this object from the backing store.  Attrs can be set or
      * deleted.  @note the attribute APIs may be revisited in the future. */
-    virtual int set_obj_attrs(const DoutPrefixProvider* dpp, Attrs* setattrs, Attrs* delattrs, optional_yield y) = 0;
+    virtual int set_obj_attrs(const DoutPrefixProvider* dpp, Attrs* setattrs, Attrs* delattrs, optional_yield y, ceph::real_time set_mtime = ceph::real_clock::zero()) = 0;
```

Comment:
please avoid using default arguments on virtual functions. i know there are other examples in rgw_sal.h, but our [style guide](https://google.github.io/styleguide/cppguide.html#Default_Arguments) argues against them:

> The defaults for arguments in a virtual function call are determined by the static type of the target object, and there's no guarantee that all overrides of a given function declare the same defaults.

> Default arguments are banned on virtual functions, where they don't work properly